extra/src/main/java/aima/extra/environment/model/map/EnvironmentMap.java

```diff
@@ -0,0 +1,199 @@
+package aima.extra.environment.model.map;
+
+/**
+ * Created by @AdrianBZG on 2/04/17.
+ */
+
+import aima.core.util.datastructure.Tuple;
+import aima.extra.environment.model.map.generator.IGenerator;
+import aima.extra.environment.model.map.object.Block;
+import aima.extra.environment.model.map.object.MapObject;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+/**
+ * Represents a full map extension
+ *
+ * TODO: Map size limitation is Int size (Use BigInteger?)
+ * TODO: Test which should be the chunk size to be more efficient in memory
+ * TODO: Map IO
+ * TODO: Unload unused MapChunk from memory to disk
+ *
+ */
+public class EnvironmentMap implements Cloneable {
+
+    private static int CHUNK_SIZE = 32;
+
+    /**
+     * Partial Map with zones explored
+     */
+    HashMap<MapSector, MapChunk> map = new HashMap<>();
+
+    Optional<Tuple<Integer, Integer>> dimensions = Optional.empty();
```

Comment:
using `Optional` as properties is considered bad practice, could you do anything else here? Like using `null`?
https://stackoverflow.com/a/29033935/1359903
https://www.insaneprogramming.be/article/2017/01/06/the-usages-of-optional/